submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestFinalizer.kt

```diff
@@ -0,0 +1,53 @@
+package ac.uk.ebi.biostd.submission.submitter.request
+
+import ac.uk.ebi.biostd.persistence.common.model.RequestStatus.PROCESSED
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestPersistenceService
+import ac.uk.ebi.biostd.persistence.filesystem.api.FileStorageService
+import ebi.ac.uk.extended.model.ExtFile
+import ebi.ac.uk.extended.model.ExtSubmission
+import mu.KotlinLogging
+import uk.ac.ebi.extended.serialization.service.ExtSerializationService
+import uk.ac.ebi.extended.serialization.service.fileSequence
+
+private val logger = KotlinLogging.logger {}
+
+class SubmissionRequestFinalizer(
+    private val storageService: FileStorageService,
+    private val serializationService: ExtSerializationService,
+    private val queryService: SubmissionPersistenceQueryService,
+    private val requestService: SubmissionRequestPersistenceService,
+) {
+    fun finalizeRequest(accNo: String, version: Int): ExtSubmission {
+        val request = requestService.getPersistedRequest(accNo, version)
+        val sub = queryService.getExtByAccNo(accNo, includeFileListFiles = true)
+        val previous = queryService.findLatestInactiveByAccNo(accNo, includeFileListFiles = true)
+
+        if (previous != null) {
+            if (sub.storageMode == previous.storageMode) deleteRemainingFiles(sub, previous)
```

Comment:
Yes, you're right in principle. However, doing it like this lets us do two things:
1. Being consistent with the approach of cleaning common -> persisting -> deleting remaining. Have in mind that we don't do anything during this stage if the storage mode are different
2. We don't need to worry about md5 or path comparison since we know for sure that, if the storage modes are different, we just need to delete the whole thing.

Reply:
Sorry not following your points here:
1. Operation that will remain is `deleteRemainingFiles` this is still consistent. Not change in flow so not sure what is your point there. 
2. Yes we do not need to worry about md5 of path but we need two opeartions instead of one.  Rather I will prefer a single function a single flow that specializations based on conditions. With the same principle is that I try we only allow submit and not others operations. 

Aditionally Trusting the submission intead of checking the files may be risky as if a file is in diferent storage (which should not happen unless we save pagetab in nfs always) we will introduce inconsistencies. Clearly we can come back to this code and changed it but I think it can be single flow and error proobe from now.

Reply:
Got it, I think I see your point now. I'll include it in the current PR